packages/fiori-freestyle-writer/src/packageConfig.ts

```diff
@@ -63,6 +66,7 @@ export function getPackageJsonTasks({
             ? {
                   'start-mock': mockTask
               }
-            : {}
+            : {},
+        addTypeScript ? { prestart: 'npm run ts-typecheck', prebuild: 'npm run ts-typecheck' } : {}
```

Comment:
Don't add the scripts programmatically but add them to the typescript template at https://github.com/SAP/open-ux-tools/blob/feat/746/support-ts-sapui5-apps/packages/ui5-application-writer/templates/optional/typescript/package.json#L3. Please also don't use `npm...` but use the same script as use for the `ts-typecheck` script so that generated apps can also be used with `yarn` or `pnpm` or ...

Reply:
@devinea which ones do have problems? Should we fix those instead?

Reply:
@tobiasqueck I haven't checked today. But from what I recall type checking is failing on FE V2 templates as we are waiting for more types to be exposed for `sap.ui.generic.AppComponent` 
Also, we probably need more tests to validate the types for the templates. 
The plan is for now to enable the type checking for the simple template. Then add the others in another iteration.
Obviously the `ts-typecheck` is still available to run manually in the meantime.

Reply:
I don't think i've seen problems using `npm run` with yarn or pnpm. 
Are you suggesting that we have this instead?
```JSON
{
"ts-typecheck": "tsc --noEmit",
"prestart": "tsc --noEmit",
"prebuild": "tsc --noEmit"
}
```

Reply:
Yes, that was my suggestion. I actually assumed it doesn't work if you'd do in a monorepo where the modules might not be in `./node_modules` but to be honest I didn't check.

Reply:
There is duplication. But, it is shorter than using `npm run ts-typecheck`  so 👍 .

Reply:
>But from what I recall type checking is failing on FE V2 templates as we are waiting for more types to be exposed for sap.ui.generic.AppComponent
Also, we probably need more tests to validate the types for the templates.

Ok, I give in here, but create a follow up issue with the requirement to enable this for all templates with the tasks
- remove conditional code added in this PR
- enhancing the template in `ui5-application-writer`
- validate and fix all provided ts templates in `fiori-*-writer` to throw no errors when running a typecheck